app/main/routes.py

```diff
@@ -162,4 +163,54 @@ def search():
 def user_popup(username):
     user = User.query.filter_by(username=username).first_or_404()
     form = EmptyForm()
-    return render_template('user_popup.html', user=user, form=form)
\ No newline at end of file
+    return render_template('user_popup.html', user=user, form=form)
+
+
+@main_blueprint.route('/send_message/<recipient>', methods=['GET', 'POST'])
+@login_required
+def send_message(recipient):
+    user = User.query.filter_by(username=recipient).first_or_404()
+    form = MessageForm()
+    if form.validate_on_submit():
+        user.add_notification('unread_message_count', user.new_messages_count())
```

Comment:
we should first add message to the database and only after that create notification
```suggestion
          msg = Message(author=current_user, recipient=user,
                    body=form.message.data)
          db.session.add(msg)
          user.add_notification('unread_message_count', user.new_messages_count())
          db.session.commit()
```
Pay attention that we can use one `commit` statement